### ISSUE TITLE
Garbage collection breaks custom mapping

### ISSUE THREAD

**Issue by 79it** [mganss/ExcelMapper#14]
Hello,

We need to comply with a mandated requirement of quote prefixing any cell staring with =, +, -, $ or @ to mitigate the risk of Excel Injection.

The application is written in .Net Core and the Excel file is generated a from an API call.

I have implemented this by adding a mapping and it works well until the garbage collection runs.
At this point 
````var columnIndex = !hasColumnsByIndex ? j : columnsByIndex.First(c => c.Value == getter.Value).Key;```` on line 438 of ExcelMapper.cs throws an exception because the _Sequence contains no matching element
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)_

The code is comparing the ColumnInfo objects and while they look the same in a debug session they don't match and therefore they're not returned from Linq query. I can get this working reliably by changing the code to compare the property names.

Is there a way to override GenerateCellSetter? I also tried providing a custom TypeMapperFactory but couldn't get that working and it looked like I'd need to duplicate half of the code anyway.

If I created a PR with a new global setting and attribute would you consider merging it into the codebase?

The setting/attribute might be something like````QuotePrefixSuspectValues```` and if this is set then it uses the following to set the value:

````
var injectionCharacters = new[] { '=', '-', '+', '@', '%' };
c.SetCellValue(o.ToString().IndexOfAny(injectionCharacters) > -1 ? $"'{o}" : o.ToString());
````

What do you think?

Thanks

Chris

**Comment by mganss** [comment]
Can you post or attach a short example that shows the problem?

**Comment by 79it** [comment]
[ExcelMapperProblem.zip](https://github.com/mganss/ExcelMapper/files/3028864/ExcelMapperProblem.zip)

GC.Collect() is only there to speed it up - in my application the problem manifests itself after 5 - 10 minutes.

The first run works, however, subsequent runs will fail.

Thanks

Chris

**Comment by mganss** [comment]
`GC.Collect()` wasn't even necessary, the exception occurred on the second call to `Save()`. This was because when on each call to `AddMapping` a new `ColumnInfo` object was created which didn't then match the cached object from the previous run. Fixed in 5.0.46.
